runtime/lua/vim/_comment.lua

```diff
@@ -9,8 +9,8 @@
 local function get_commentstring(ref_position)
   local buf_cs = vim.bo.commentstring
 
-  local has_ts_parser, ts_parser = pcall(vim.treesitter.get_parser)
-  if not has_ts_parser then
+  local ts_parser = vim.treesitter._get_parser()
```

Comment:
No, the point is to get rid of pcall. It's not the right idiom for this.

Reply:
The `pcall()` will be used not to "test if there is parser", but to "avoid any error at all cost". Because working with tree-sitter is not the main priority here. But again, this is mostly a style preference.

Reply:
> "avoid any error at all cost".

What error could there be?

Reply:
> What error could there be?

An error not known in advance. Which is how errors commonly appear.

Reply:
Does that mean we will have to start wrapping _every_ function in `pcall`? No. Either there is a parser, in which case it's returned, or there is not (for whatever reason), in which case `nil` is returned. Any errors that arise during the check are the responsibility of `get_parser` and should be handled there.

Reply:
My point was that the act of commenting does not directly linked to tree-sitter. So avoiding any errors from it for a graceful fallback would be my choice here.

However (for the third time), this is a style preference. Meaning I don't really have strong opinion on this (otherwise it would have been stated that way) and defer the choice to the author.

Reply:
My point is to clearly state -- especially for future reference -- the official style (avoid pcall within our own sphere of influence in favor of returning `nil` if possible -- pcall is for guarding _external_ functions you cannot control, but here we control the whole code path).